### PR TITLE
update reccomended way to load dependencies

### PR DESCRIPTION
**Pull request by ggggggggg**
Reflecting #196, stop recommending `@load_dependencies` since it doesn't work. Instead provide a few lines of code taken from HDF5.jl that will load deps.jl. [ci skip]

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -254,42 +254,16 @@ provides(Sources,URI("http://libvirt.org/sources/libvirt-1.1.1-rc2.tar.gz"),libv
...
-@BinDeps.load_dependencies
-```
-
-This will make all your libraries available as variables named by the names you gave 
-the dependency. E.g. if you declared a dependency as
-
+To load dependencies without a runtime dependence on BinDeps, place code like the following near the start of the module's primary file. Don't forget to change the error message to reflect the name of your package.
 ```julia
-	library_dependency("libfoo")
+const depsfile = joinpath(dirname(@__DIR__), "deps", "deps.jl")
```
Comment (tkelman) [inline README.md]: doing `dirname(dirname(@__FILE__))` instead of `dirname(@__DIR__)` would avoid a Compat dependency and work on 0.5

Comment (tkelman) [inline README.md]: it is one level different though, so would need another dirname or another `"..",` argument

**Comment by ggggggggg** [comment]
Fixed your comment and added back in the description which variables the dependencies are loaded into.

File: README.md [hunk 2]
```diff
@@ -254,41 +254,25 @@ provides(Sources,URI("http://libvirt.org/sources/libvirt-1.1.1-rc2.tar.gz"),libv
...
-
-if you do not want to change the names of the variables that these libraries get 
-stored in, you may use
-
+To load dependencies without a runtime dependence on BinDeps, place code like the following
+near the start of the Package's primary file. Don't forget to change the error message to 
+reflect the name of the package.
 ```julia
-@BinDeps.load_dependencies [:libfoo=>:_foo, :libbar=>:_bar]
+const depsfile = joinpath(dirname(@__FILE__), "deps", "deps.jl")
```
Comment (tkelman) [inline README.md]: should either call `dirname` twice or do `dirname(@__FILE__), "..", "deps", "deps.jl"`

**Comment by ggggggggg** [comment]
Thanks. I actually tested it this time, and it seems to work.

File: README.md [hunk 3]
```diff
@@ -254,42 +254,26 @@ provides(Sources,URI("http://libvirt.org/sources/libvirt-1.1.1-rc2.tar.gz"),libv
...
-The basic usage is very simple:
-
-```jl
-using BinDeps
-@BinDeps.load_dependencies
+To load dependencies without a runtime dependence on BinDeps, place code like the following
+near the start of the Package's primary file. Don't forget to change the error message to 
+reflect the name of the package.
+```julia
+const depsfile = joinpath(dirname(@__FILE__), "..", "deps", "deps.jl")
```
Comment (musm) [inline README.md]: better would be `joinpath(dirname(@__DIR__), "deps", "deps.jl")`   which needs Compat

Comment (tkelman) [inline README.md]: see previous versions, needing Compat makes it not better

Comment (musm) [inline README.md]: I see now. But isn't it best practice to encourage the latest syntax and let Compat take care of backwards compatibility.

Comment (tkelman) [inline README.md]: Only if people want to use Compat, which not everyone does.

File: README.md [hunk 4]
```diff
@@ -254,42 +254,26 @@ provides(Sources,URI("http://libvirt.org/sources/libvirt-1.1.1-rc2.tar.gz"),libv
...
+if isfile(depsfile)
+    include(depsfile)
+else
+    error("HDF5 not properly installed. Please run Pkg.build(\"HDF5\")")
```
Comment (musm) [inline README.md]: `Please run ....  then restart Julia.`

**Comment by ggggggggg** [comment]
updated

**Comment by musm** [comment]
bump

**Comment by musm** [comment]
Thanks!

**Pull request merged**